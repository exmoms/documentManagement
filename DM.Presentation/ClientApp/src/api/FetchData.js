/**
 * @file this file includes functions that fetch data from API
 */

// Generic Delete Method
export async function deleteFromApi(url) {
  let res = await fetch(url, {
    method: "DELETE",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  });
  return res;
}

// Get MetaData Models
export async function fetchData(url) {
  let object = {};
  /*
   * send the request through a CORS proxy
   * https://stackoverflow.com/questions/43262121/trying-to-use-fetch-and-pass-in-mode-no-cors/43268098#43268098
   */
  await fetch(url, {
    method: "GET",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  })
    .then((response) => response.json())
    .then(
      (data) => (object = data),
      (error) => console.log(error)
    );
  return object;
}
