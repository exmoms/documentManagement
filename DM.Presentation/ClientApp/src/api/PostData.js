/**
 * @file This file includes functions that help to post data to a specific API
 * @version 1.0
 */

/**
 * Post new document data.
 * Upload form-data, scanned copy of doc and attachments.
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @param {*} documentData
 * @param {*} images
 * @param {*} attachments
 * @param {*} url
 * @param {*} option its values either "Add" for add new document version or "Update" to update exsiting document Version
 */
export async function postDocumentData(
  documentData,
  scannedPages,
  attachments,
  url = "",
  option
) {
  var formData = new FormData();

  if (process.env.NODE_ENV === "development") {
    console.log(documentData);
    console.log(attachments);
    console.log(scannedPages);
  }

  switch (option) {
    case "Add":
      formData.append("Document", JSON.stringify(documentData));
      break;
    case "Update":
      formData.append("Version", JSON.stringify(documentData));
      break;
    default:
  }

  attachments.forEach((attachment) => {
    formData.append("Attachments", attachment);
  });
  scannedPages.forEach((scannedPage) => {
    formData.append("Scans", scannedPage);
  });

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    headers: {
      Accept: "application/json",
      // "Content-Type": "application/json",
    },
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Attachment Data.
 * Upload attachment json data + attachment file  .
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} payload
 * @param {*} attachmentFile
 * @param {*} url
 */
export async function postAttachmentData(payload, attachmentFile, url = "") {
  var formData = new FormData();

  formData.append("Attachment", JSON.stringify(payload));
  formData.append("FileAttachment", attachmentFile);

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Request To API
 * Send post request to api with  payload  and return a response.
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} url The api url
 * @param {*} payload js object which will be sent as request body.
 */
export async function postDataToAPI(url, payload = null, method = "POST") {

  const response = await fetch(url, {
    method: method,
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(payload),
  });
  return response;
}
