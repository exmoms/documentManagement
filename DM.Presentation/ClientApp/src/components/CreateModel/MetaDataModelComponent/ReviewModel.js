import React from "react";
import Typography from "@material-ui/core/Typography";
import ReactVirtualizedTable from "./TableView";
import { useTranslation } from "react-i18next";
import { makeStyles } from "@material-ui/core";

const useStyle = makeStyles(() => ({
  textAlign: {
    textAlign: "initial",
    padding: "20px",
  },
}));

export default function ReviewModel(props) {
  const { t } = useTranslation();
  const classes = useStyle();

  const attributeColumns = [
    { id: "metaDataAttributeName", label: t("nameofattribute"), minWidth: 170 },
    { id: "dataTypeID", label: t("type"), minWidth: 100 },
    {
      id: "isRequired",
      label: t("isrequired"),
      minWidth: 170,
      align: "center",
    },
  ];

  const selectedMetaDataModels_aggregateMetaDataModelsParts_Columns = [
    { id: "aggregateName", label: "Aggregation Name", minWidth: 170 },
    { id: "metaDataModelName", label: "Aggregated Model", minWidth: 100 },
  ];

  const compound_Columns = [
    { id: "caption", label: "caption", minWidth: 170 },
    { id: "isRequired", label: t("isrequired"), minWidth: 100 },
  ];

  var selectedMetaDataModels_aggregateMetaDataModelsParts = [
    {
      metaDataModelName: "",
      aggregateName: "",
    },
  ];

  if (
    props.value.selectedMetaDataModels.length ===
    props.value.aggregateMetaDataModelsParts.length
  ) {
    for (let i = 0; i < props.value.aggregateMetaDataModelsParts.length; i++) {
      selectedMetaDataModels_aggregateMetaDataModelsParts[i] = {
        metaDataModelName:
          props.value.selectedMetaDataModels[i].metaDataModelName,
        aggregateName:
          props.value.aggregateMetaDataModelsParts[i].aggregateName,
      };
    }
  }

  let compound;
  let aggregate;
  if (props.value.isCompound === true) {
    compound = (
      <>
        <p id="isCompound">
          <b>{t("compoundmodelincludes")}</b>
        </p>
        <ReactVirtualizedTable
          rows={props.value.compoundModels}
          columns={compound_Columns}
        />
      </>
    );
  } else {
    compound = (
      <p id="isCompound">
        <b>{t("notcompoundmodel")}</b>
      </p>
    );
  }

  if (props.value.isAggregated === true) {
    aggregate = (
      <>
        <p id="isAggregated">
          <b>{t("aggregatedmodelincludes")}</b>
        </p>
        <ReactVirtualizedTable
          rows={selectedMetaDataModels_aggregateMetaDataModelsParts}
          columns={selectedMetaDataModels_aggregateMetaDataModelsParts_Columns}
        />
      </>
    );
  } else {
    aggregate = (
      <p id="isAggregated">
        <b>{t("notaggregated")}</b>
      </p>
    );
  }

  const types = [
    t("booltype"),
    t("datetype"),
    t("decimal"),
    t("double"),
    t("int"),
    t("string"),
  ];

  var Attributes = [];

  for (let i = 0; i < props.value.metaDataAttribute.length; i++) {
    Attributes.push({
      dataTypeID: types[props.value.metaDataAttribute[i].dataTypeID - 1],
      metaDataAttributeName:
        props.value.metaDataAttribute[i].metaDataAttributeName,
      isRequired: props.value.metaDataAttribute[i].isRequired,
    });
  }

  return (
    <React.Fragment>
      <div className={classes.textAlign}>
        <Typography variant="h6" gutterBottom align="center">
          {t("modelreview")}
        </Typography>
        <hr />
        <p id="ModelName">
          <b>{t("modelname")}</b>: {props.value.metaDataModelName}
        </p>
        <p id="DocumentClassName">
          <b>{t("documentparagraph")}</b>:{" "}
          {props.value.documentClassSelected.documentClassName}
        </p>
        {compound}
        <br />
        {aggregate}
        <br />
        <b>{t("paragraphmetadata")}</b>:
        <br />
        <ReactVirtualizedTable rows={Attributes} columns={attributeColumns} />
      </div>
    </React.Fragment>
  );
}
