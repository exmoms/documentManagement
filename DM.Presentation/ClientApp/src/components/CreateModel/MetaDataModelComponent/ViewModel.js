import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import MuiDialogActions from "@material-ui/core/DialogActions";
import MuiDialogTitle from "@material-ui/core/DialogTitle";
import IconButton from "@material-ui/core/IconButton";
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import CloseIcon from "@material-ui/icons/Close";
import React, { useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import { fetchData } from "../../../api/FetchData";
import ReviewModel from "./ReviewModel";
const styles = (theme) => ({
  root: {
    margin: 0,
    padding: theme.spacing(2),
  },
  closeButton: {
    position: "absolute",
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  },
});

const DialogTitle = withStyles(styles)((props) => {
  const { children, classes, onClose, ...other } = props;

  return (
    <MuiDialogTitle disableTypography className={classes.root} {...other}>
      <Typography variant="h6">{children}</Typography>
      {onClose ? (
        <IconButton
          aria-label="close"
          className={classes.closeButton}
          onClick={onClose}
        >
          <CloseIcon />
        </IconButton>
      ) : null}
    </MuiDialogTitle>
  );
});

const DialogActions = withStyles((theme) => ({
  root: {
    margin: 0,
    padding: theme.spacing(1),
  },
}))(MuiDialogActions);

export default function CustomizedDialogs(props) {
  const [metaDataModelName, setMetaDataModelName] = useState("");
  const [documentClassSelected, setDocumentClassSelected] = useState({
    documentClassName: "",
  });

  const [isCompound, setIsCompound] = useState(false);
  const [isAggregated, setIsAggregated] = useState(false);
  const [
    aggregateMetaDataModelsPart,
    setAggregateMetaDataModelsParts,
  ] = useState([{ aggregateName: "" }]);
  const [compoundModels, setCompoundModels] = useState([
    { isRequired: false, caption: "" },
  ]);
  const [selectedMetaDataModels, setSelectedMetaDataModel] = useState([
    { metaDataModelName: "" },
  ]);
  const [metaDataAttribute, setmetaDataAttribute] = useState([
    {
      metaDataAttributeName: "",
      isRequired: false,
      dataTypeID: "",
    },
  ]);

  useEffect(() => {
    if (props.id !== -1) {
      fetchData(`/api/MetaDataModel/${props.id}`)
        .then((value) => {
          setMetaDataModelName(value.metaDataModelName);
          setDocumentClassSelected({
            documentClassName: value.documentClassName,
          });

          setIsCompound(value.compoundModels.length !== 0 ? true : false);
          setIsAggregated(
            value.childMetaDataModels.length !== 0 ? true : false
          );
          if (
            value.childMetaDataModels &&
            value.childMetaDataModels.length !== 0
          ) {
            var selected_models = [];
            for (var i = 0; i < value.childMetaDataModels.length; i++) {
              selected_models.push({
                metaDataModelName:
                  value.childMetaDataModels[i].childMetaDataModelName,
              });
            }
            setSelectedMetaDataModel(selected_models);
            setAggregateMetaDataModelsParts(value.childMetaDataModels);
          }
          if (value.compoundModels) {
            setCompoundModels(value.compoundModels);
          }

          setmetaDataAttribute(value.metaDataAttributes);
        })
        .catch((e) => console.log(e));
    }
  }, [props.id]);

  const { t } = useTranslation();
  return (
    <Dialog
      onClose={props.handleClose}
      aria-labelledby="customized-dialog-title"
      open={props.open}
    >
      <DialogTitle id="customized-dialog-title" onClose={props.handleClose} />

      <ReviewModel
        value={{
          metaDataModelName: metaDataModelName,
          documentClassSelected: documentClassSelected,
          isCompound: isCompound,
          isAggregated: isAggregated,
          aggregateMetaDataModelsParts: aggregateMetaDataModelsPart,
          compoundModels: compoundModels,
          selectedMetaDataModels: selectedMetaDataModels,
          metaDataAttribute: metaDataAttribute,
        }}
      />

      <DialogActions>
        <Button
          autoFocus
          onClick={() => {
            window.print();
          }}
          color="primary"
        >
          {/* print */}
          {t("print")}
        </Button>
      </DialogActions>
    </Dialog>
  );
}
