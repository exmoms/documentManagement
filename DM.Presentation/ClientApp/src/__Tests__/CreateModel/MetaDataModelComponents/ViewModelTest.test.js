import React from "react";
import ReactDom from "react-dom";
import ViewModel from "../../../components/CreateModel/MetaDataModelComponent/ViewModel.js";
import { cleanup } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import { act } from "react-dom/test-utils";
import * as FetchDataMock from "../../../api/FetchData";
import ReviewModel from "../../../components/CreateModel/MetaDataModelComponent/ReviewModel";
import { getModelById } from "../../../api/FetchData";
afterEach(cleanup);
describe("ViewModel", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  test("renders without crashing", () => {
    let object = {
      metaDataModelId: 3,
      metaDataModelName: "Model A",
      documentClassId: 4,
      documentClassName: "Class A",
      metaDataAttributes: [
        {
          metaDataAttributeId: 1,
          metaDataAttributeName: "A bool",
          isRequired: true,
          dataTypeID: 1,
          dataTypeName: "bool",
        },
        {
          metaDataAttributeId: 2,
          metaDataAttributeName: "A int",
          isRequired: true,
          dataTypeID: 5,
          dataTypeName: "int",
        },
        {
          metaDataAttributeId: 3,
          metaDataAttributeName: "A decimal",
          isRequired: false,
          dataTypeID: 3,
          dataTypeName: "decimal",
        },
      ],
        aggregateMetaDataModelsParts: [{  metaDataModelName: "" }],
      compoundModels: [
        {
          compoundModelsId: 1,
          metadataModelId: 3,
          isRequired: true,
          caption: "attach name 1",
        },
        {
          compoundModelsId: 2,
          metadataModelId: 3,
          isRequired: false,
          caption: "attach name 2",
        },
      ],
      addedDate: "2020-02-25T15:55:42.34",
    };
    const mockSuccessResponse = object;
    const mockJsonPromise = Promise.resolve(mockSuccessResponse); // 2
    const FetchMock = jest
      .spyOn(FetchDataMock, "fetchData")
      .mockImplementation(() => mockJsonPromise);
    const div = document.createElement("div");
    let idMock = 0;
    const handleCloseMock = jest.fn();
    let open = true;
    ReactDom.render(
      <ViewModel id={idMock} handleClose={handleCloseMock} open={open} />,
      div
    );
  });

  test("matches snapshot", () => {
    let object = {
      metaDataModelId: 3,
      metaDataModelName: "Model A",
      documentClassId: 4,
      documentClassName: "Class A",
      metaDataAttributes: [
        {
          metaDataAttributeId: 1,
          metaDataAttributeName: "A bool",
          isRequired: true,
          dataTypeID: 1,
          dataTypeName: "bool",
        },
        {
          metaDataAttributeId: 2,
          metaDataAttributeName: "A int",
          isRequired: true,
          dataTypeID: 5,
          dataTypeName: "int",
        },
        {
          metaDataAttributeId: 3,
          metaDataAttributeName: "A decimal",
          isRequired: false,
          dataTypeID: 3,
          dataTypeName: "decimal",
        },
      ],
        aggregateMetaDataModelsParts: [{ aggregateName: "agregated" }],
      compoundModels: [
        {
          compoundModelsId: 1,
          metadataModelId: 3,
          isRequired: true,
          caption: "attach name 1",
        },
        {
          compoundModelsId: 2,
          metadataModelId: 3,
          isRequired: false,
          caption: "attach name 2",
        },
      ],
      addedDate: "2020-02-25T15:55:42.34",
    };
    const mockSuccessResponse = object;
    const mockJsonPromise = Promise.resolve(mockSuccessResponse); // 2
    const FetchMock = jest
      .spyOn(FetchDataMock, "fetchData")
      .mockImplementation(() => mockJsonPromise);
    const div = document.createElement("div");
    let idMock = 0;
    const handleCloseMock = jest.fn();
    let open = true;
    const wrapper = shallow(
      <ViewModel id={idMock} handleClose={handleCloseMock} open={open} />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

    test("first render of ReviewModel", () => {
    let object = {
      metaDataModelId: 1,
      metaDataModelName: "Model A",
      documentClassId: 4,
      documentClassName: "Class A",
      metaDataAttributes: [
        {
              dataTypeID: 3,
              id: 3098,
              isRequired: true,
              metaDataAttributeName: "fdes",
        },
        {
            dataTypeID: 6,
            id: 3099,
            isRequired: true,
            metaDataAttributeName: "sdaa",
        }
      ],
        aggregateMetaDataModelsParts: [{
            aggregateName: "errrrr",
            childMetaDataModelId: 2024,
            childMetaDataModelName: "dsds",
            id: 3053,
            parentMetaDataModelId: 4076,
        }],
      compoundModels: [
        {
            caption: "wewq",
            id: 3055,
            isRequired: true,
            metadataModelId: 4076,
        },
        {
            caption: "dasdas",
            id: 3056,
            isRequired: false,
            metadataModelId: 4076,
        }
      ],
      addedDate: "2020-02-25T15:55:42.34",
    };
    const mockSuccessResponse = object;
    const mockJsonPromise = Promise.resolve(mockSuccessResponse); // 2
    const fetchMock = jest
      .spyOn(FetchDataMock, "fetchData") 
      .mockImplementation(() => mockJsonPromise);
      let idMock = -1;
    const handleCloseMock = jest.fn();
    let wrapper;
    return mockJsonPromise.then(() => {
       wrapper = mount(
        <ViewModel id={idMock} handleClose={handleCloseMock} open={open} />
      );
    }).then(() => {
      wrapper.render();
      wrapper.update(); 
      expect(wrapper.find(ReviewModel).props().value.metaDataModelName).toBe(
      ""
    );
    expect(wrapper.find(ReviewModel).props().value.documentClassSelected).toStrictEqual(
      {"documentClassName": ""}
        ); 
        expect(wrapper.find(ReviewModel).props().value.isCompound).toBeFalsy();
    expect(wrapper.find(ReviewModel).props().value.isAggregated).toBeFalsy();
    expect(
      wrapper.find(ReviewModel).props().value.aggregateMetaDataModelsParts
    ).toStrictEqual([{ "aggregateName": ""  }]);
    expect(wrapper.find(ReviewModel).props().value.compoundModels).toStrictEqual([
        {
            isRequired: false,
            caption: ""
        }
    ]);
    expect(wrapper.find(ReviewModel).props().value.metaDataAttribute).toStrictEqual([
      {
        isRequired: false,
        dataTypeID: "",
        metaDataAttributeName: "",
      },
    ]);
    expect(fetchMock).toHaveBeenCalledTimes(0);
    });
    
    
  });

    test("Get Model by id", () => {
        let object = {
            id: 4076,
            metaDataModelName: "rytr",
            documentClassId: 1002,
            documentClassName: "Doc_class_3",
            addedDate: "0001-01-01T00:00:00",
            userId: 1,
            metaDataAttributes: [
                {
                    id: 3098,
                    metaDataAttributeName: "fdes",
                    isRequired: true,
                    dataTypeID: 3
                },
                {
                    id: 3099,
                    metaDataAttributeName: "sdaa",
                    isRequired: true,
                    dataTypeID: 6
                },
                {
                    id: 3100,
                    metaDataAttributeName: "cvxxx",
                    isRequired: false,
                    dataTypeID: 6
                }
            ],
            childMetaDataModels: [
                {
                    id: 3053,
                    childMetaDataModelId: 2024,
                    parentMetaDataModelId: 4076,
                    aggregateName: "errrrr",
                    childMetaDataModelName: "dsds"
                },
                {
                    id: 3054,
                    childMetaDataModelId: 2026,
                    parentMetaDataModelId: 4076,
                    aggregateName: "rreeee",
                    childMetaDataModelName: "Model 21"
                },
                {
                    id: 3055,
                    childMetaDataModelId: 3035,
                    parentMetaDataModelId: 4076,
                    aggregateName: "wwweee",
                    childMetaDataModelName: "sdsds"
                }
            ],
            compoundModels: [
                {
                    id: 3055,
                    metadataModelId: 4076,
                    isRequired: true,
                    caption: "wewq"
                },
                {
                    id: 3056,
                    metadataModelId: 4076,
                    isRequired: false,
                    caption: "dasdas"
                },
                {
                    id: 3057,
                    metadataModelId: 4076,
                    isRequired: false,
                    caption: "cvxdd"
                }
            ]
        };
        const mockSuccessResponse = object;
        const mockJsonPromise = Promise.resolve(mockSuccessResponse);
        const FetchMock = jest
            .spyOn(FetchDataMock, "fetchData")
            .mockImplementation(() => mockJsonPromise);
        let idMock = 4075;
        const handleCloseMock = jest.fn();
        let wrapper;
        return mockJsonPromise.then(() => {
            wrapper = mount(
                <ViewModel id={idMock} handleClose={handleCloseMock} open={open} />
            );
        }).then(() => {
            wrapper.render();
            wrapper.update();
            //expect the side effects  
            expect(FetchMock).toHaveBeenCalledTimes(1);
            expect(FetchMock.mock.calls[0][0]).toBe('/api/MetaDataModel/'+idMock);
            expect(wrapper.find(ReviewModel).props().value.metaDataModelName).toBe(
                "rytr"
            );
            expect(wrapper.find(ReviewModel).props().value.documentClassSelected).toStrictEqual(
                { "documentClassName": "Doc_class_3" }
            );
            expect(wrapper.find(ReviewModel).props().value.isCompound).toBeTruthy();
            expect(wrapper.find(ReviewModel).props().value.isAggregated).toBeTruthy();
            expect(
                wrapper.find(ReviewModel).props().value.aggregateMetaDataModelsParts
            ).toStrictEqual([
                { 
                aggregateName: "errrrr",
                childMetaDataModelId: 2024,
                childMetaDataModelName: "dsds",
                id: 3053,
                parentMetaDataModelId: 4076,
                },
                {
                    aggregateName: "rreeee",
                    childMetaDataModelId: 2026,
                    childMetaDataModelName: "Model 21",
                    id: 3054,
                    parentMetaDataModelId: 4076,
                },
                {
                    aggregateName: "wwweee",
                    childMetaDataModelId: 3035,
                    childMetaDataModelName: "sdsds",
                    id: 3055,
                    parentMetaDataModelId: 4076,
                }
            ]);
            expect(wrapper.find(ReviewModel).props().value.compoundModels).toStrictEqual([
                {
                    caption: "wewq",
                    id: 3055,
                    isRequired: true,
                    metadataModelId: 4076,
                },
                {
                    caption: "dasdas",
                    id: 3056,
                    isRequired: false,
                    metadataModelId: 4076,
                },
                {
                    caption: "cvxdd",
                    id: 3057,
                    isRequired: false,
                    metadataModelId: 4076,

                }
            ]);
            expect(wrapper.find(ReviewModel).props().value.metaDataAttribute).toStrictEqual([
                {
                    id: 3098,
                    metaDataAttributeName: "fdes",
                    isRequired: true,
                    dataTypeID: 3
                },
                {
                    id: 3099,
                    metaDataAttributeName: "sdaa",
                    isRequired: true,
                    dataTypeID: 6
                },
                {
                    id: 3100,
                    metaDataAttributeName: "cvxxx",
                    isRequired: false,
                    dataTypeID: 6
                }
            ]);
        });


    });
});

