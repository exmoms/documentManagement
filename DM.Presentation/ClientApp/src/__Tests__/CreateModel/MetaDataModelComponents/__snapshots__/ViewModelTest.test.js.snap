// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewModel matches snapshot 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-labelledby="customized-dialog-title"
  onClose={[MockFunction]}
  open={true}
>
  <WithStyles(Component)
    id="customized-dialog-title"
    onClose={[MockFunction]}
  />
  <ReviewModel
    value={
      Object {
        "aggregateMetaDataModelsParts": Array [
          Object {
            "aggregateName": "",
          },
        ],
        "compoundModels": Array [
          Object {
            "caption": "",
            "isRequired": false,
          },
        ],
        "documentClassSelected": Object {
          "documentClassName": "",
        },
        "isAggregated": false,
        "isCompound": false,
        "metaDataAttribute": Array [
          Object {
            "dataTypeID": "",
            "isRequired": false,
            "metaDataAttributeName": "",
          },
        ],
        "metaDataModelName": "",
        "selectedMetaDataModels": Array [
          Object {
            "metaDataModelName": "",
          },
        ],
      }
    }
  />
  <WithStyles(WithStyles(ForwardRef(DialogActions)))>
    <WithStyles(ForwardRef(Button))
      autoFocus={true}
      color="primary"
      onClick={[Function]}
    >
      print
    </WithStyles(ForwardRef(Button))>
  </WithStyles(WithStyles(ForwardRef(DialogActions)))>
</WithStyles(ForwardRef(Dialog))>
`;
